sw/device/lib/testing/test_framework/coverage_llvm.c

```diff
@@ -19,33 +19,51 @@
 int __llvm_profile_runtime;
 
 /**
- * Sends the given buffer as a hex string over UART.
+ * Coverage buffer in `.bss` to avoid stack issues.
  */
-static void send_buffer(uint8_t *buf, size_t len) {
-  for (uint32_t i = 0; i < len; ++i) {
-    base_printf("%02X", buf[i]);
-  }
-}
+static char buf[0x8000] = {0};
 
+/**
+ * Sends the given buffer as a hex string over UART.
+ *
+ * Lines are limited to 80 characters. Longer buffers will be broken to multiple
+ * lines.
+ */
 void coverage_send_buffer(void) {
   // It looks like we don't have a way to read the profile buffer incrementally.
   // Thus, we define the following buffer.
-  uint8_t buf[8192] = {0};
   // Write the profile buffer to the buffer defined above.
-  uint32_t buf_size = (uint32_t)__llvm_profile_get_size_for_buffer();
-  if (buf_size > sizeof(buf)) {
-    LOG_ERROR("ERROR: LLVM profile buffer is too large: %u bytes.", buf_size);
+  uint64_t buf_size_u64 = __llvm_profile_get_size_for_buffer();
+  if (buf_size_u64 > sizeof(buf)) {
+    LOG_ERROR("ERROR: LLVM profile buffer is too large: %u bytes.",
+              (uint32_t)buf_size);
```

Comment:
`(uint32_t)buf_size` --> `(uint32_t)buf_size_u64`?

Reply:
Ah, thanks for catching this, I forgot that I commented out the build part from my local script :)